include/jemalloc/internal/prof.h

```diff
@@ -458,16 +458,16 @@ prof_sample_accum_update(tsd_t *tsd, size_t usize, bool update,
 }
 
 JEMALLOC_ALWAYS_INLINE prof_tctx_t *
-prof_alloc_prep(tsd_t *tsd, size_t usize, bool prof_active, bool update)
+prof_alloc_prep(tsd_t *tsd, size_t usize, bool prof_unused, bool update)
 {
 	prof_tctx_t *ret;
 	prof_tdata_t *tdata;
 	prof_bt_t bt;
 
 	assert(usize == s2u(usize));
 
-	if (!prof_active || likely(prof_sample_accum_update(tsd, usize, update,
-	    &tdata)))
+	if (!malloc_option_default_on(&prof_active) ||
```

Comment:
This isn't safe, because there are code paths that require consistency re: ```prof_active``` across a series of function invocations, e.g. in ```irealloc_prof()```.  Should I just revert this particular change, or is there some other way we can accomplish this safely?